plugins/event/sqs/sqs.go

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package sqs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/hashicorp/go-secure-stdlib/awsutil"
+	"github.com/hashicorp/vault/sdk/event"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/hashicorp/vault/version"
+	"github.com/mitchellh/mapstructure"
+)
+
+var _ event.EventSubscriptionPlugin = (*sqsBackend)(nil)
+
+const pluginName = "sqs"
+
+// ErrQueueRequired is returned if the required queue parameters are not present.
+var ErrQueueRequired = errors.New("queue_name or queue_url must be specified")
+
+// New returns a new instance of the SQS plugin backend.
+func New() event.EventSubscriptionPlugin {
+	return &sqsBackend{
+		connections: map[string]*sqsConnection{},
+	}
+}
+
+type sqsBackend struct {
+	connections map[string]*sqsConnection
+	clientLock  sync.RWMutex
+}
+
+type sqsConnection struct {
+	client        *sqs.SQS
+	config        *sqsConfig
+	queueURL      string
+	ctx           context.Context
+	ctxCancelFunc func() // used when we are destroying the connection or when there is an error
+}
+
+type sqsConfig struct {
+	event.SubscribeConfigDefaults
+	CreateQueue     bool   `mapstructure:"create_queue"`
+	AccessKeyID     string `mapstructure:"access_key_id"`
+	SecretAccessKey string `mapstructure:"secret_access_key"`
+	Region          string `mapstructure:"region"`
+	QueueName       string `mapstructure:"queue_name"`
+	QueueURL        string `mapstructure:"queue_url"`
+}
+
+func newClient(sconfig *sqsConfig) (*sqs.SQS, error) {
+	var options []awsutil.Option
+	if sconfig.AccessKeyID != "" && sconfig.SecretAccessKey != "" {
+		options = append(options, awsutil.WithAccessKey(sconfig.AccessKeyID))
+		options = append(options, awsutil.WithSecretKey(sconfig.SecretAccessKey))
+	}
+	if sconfig.Region != "" {
+		options = append(options, awsutil.WithRegion(sconfig.Region))
+	}
+	options = append(options, awsutil.WithEnvironmentCredentials(true))
+	credConfig, err := awsutil.NewCredentialsConfig(options...)
+	if err != nil {
+		return nil, err
+	}
+	session, err := credConfig.GetSession()
+	if err != nil {
+		return nil, err
+	}
+	return sqs.New(session), nil
+}
+
+func (s *sqsBackend) Subscribe(_ context.Context, request *event.SubscribeRequest) error {
```

Comment:
It feels like we should probably be providing the `ctx` to the remote calls that create/query queues. Maybe that's a change that will be easier/more obvious with the v2 SDK.

Reply:
Yeah, I don't think that is possible with the v1 SDK, sadly :(